pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -1236,12 +1239,12 @@ class IsingXY(Operation):
     parameter_frequencies = [(0.5, 1.0)]
 
     def generator(self):
-        return qml.s_prod(
-            0.25,
-            qml.sum(
-                qml.prod(PauliX(wires=self.wires[0]), PauliX(wires=self.wires[1])),
-                qml.prod(PauliY(wires=self.wires[0]), PauliY(wires=self.wires[1])),
-            ),
+        return qml.Hamiltonian(
+            [0.25],
+            [
+                PauliX(wires=self.wires[0]) @ PauliX(wires=self.wires[1])
+                + PauliY(wires=self.wires[0]) @ PauliY(wires=self.wires[1])
+            ],
```

Comment:
Not sure why this one is an issue and everything else is fine, but this seems to break `parameter_frequencies` (and thereby adjoint differentiation) for `IsingXY`. 

To make sure we preserve functionality, let's add the `use_legacy_and_new_opmath` fixture to the parameter frequency test as well (getting parameter frequencies is one of the main uses for the generators). And then debug why this particular operator is failing 😅